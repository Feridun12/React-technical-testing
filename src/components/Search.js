import React, { useState } from "react";
import PropTypes from 'prop-types';
import getImages from "../requests/getImages";
import "../styles/Search.css";


function Search({ setSearchResults }) {
  // state
  const [value, setValue] = useState();
  // 
  const handleSubmit = async (event) => {
    event.preventDefault();
    setSearchResults( await getImages(value));
  };
  return (
    <div className="search-section">
      <form onSubmit={handleSubmit} className="search-form">
        <input
        placeholder="Search NASA's image library "
          onChange={(e) => setValue(e.target.value)}
          type="text"
          className="search-input"
          data-testid="search-input"
        ></input>
        <button type="submit" className="search-btn" data-testid="search-btn">
          Go!
        </button>
      </form>
    </div>
  );
}

Search.propTypes = {
 setSearchResults: PropTypes.func.isRequired,
}

export default Search;
