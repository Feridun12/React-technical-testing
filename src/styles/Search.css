.search-input {
  width: 325px;
  height: 50px;
  border-radius: 20px;
  font-size: 25px;
  padding: 0 10px;
  transition: all 0.5s ease;
  border: 1px solid white;
}

.search-input:hover {
  background-color: transparent;
  color: white;
  -webkit-box-shadow: 0px 0px 8px 1px rgba(255, 255, 255, 1);
  -moz-box-shadow: 0px 0px 8px 1px rgba(255, 255, 255, 1);
  box-shadow: 0px 0px 8px 1px rgba(255, 255, 255, 1);
}

.search-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  height: 130px;
  margin-top: 10%;
}

.search-btn {
  width: 200px;
  height: 50px;
  border-radius: 20px;
  font-size: 25px;
  color: black;
  background-color: white;
  transition: all 0.5s ease;
  border: 1px solid white;
}

.search-input::placeholder {
  text-align: center;
}

.search-btn:hover {
  -webkit-box-shadow: 0px 0px 8px 1px rgba(255, 255, 255, 1);
  -moz-box-shadow: 0px 0px 8px 1px rgba(255, 255, 255, 1);
  box-shadow: 0px 0px 8px 1px rgba(255, 255, 255, 1);
}

@media screen and (min-width: 450px) and (max-width: 1000px) {
  .search-form {
    margin-top: 50px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }
}

@media screen and (min-width: 320px) and (max-width: 425px) {
  .search-input {
    width: 90%;
  }
}
