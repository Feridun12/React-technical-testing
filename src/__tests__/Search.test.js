import { render } from "@testing-library/react";
import Search from "../components/Search";

describe("Search.js technical testing", () =>{

 const validProps = {
  setSearchResults: () => {},
 }
  test("Search.js renders correctly: snapshot testing", () => {
    const { asFragment } = render(<Search setSearchResults={validProps.setSearchResults} />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  test("tests if the search input is rendered correctly", () => {
    const {getByTestId} = render(<Search setSearchResults={validProps.setSearchResults} />);
    const searchInput = getByTestId("search-input");
    expect(searchInput).toBeTruthy();
  })

  it("renders search button as expected", () => {
    const { getByTestId } = render(<Search setSearchResults={validProps.setSearchResults} />);
    const button = getByTestId("search-btn");
    expect(button).toBeTruthy();
  });
})