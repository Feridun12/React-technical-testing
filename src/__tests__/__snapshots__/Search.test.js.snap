// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search.js technical testing Search.js renders correctly: snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="search-section"
  >
    <form
      class="search-form"
    >
      <input
        class="search-input"
        data-testid="search-input"
        placeholder="Search..."
        type="text"
      />
      <button
        class="search-btn"
        data-testid="search-btn"
        type="submit"
      >
        Go!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
